core/arch/aarch64/proc.c

```diff
@@ -38,6 +38,55 @@ static int num_simd_saved;
 static int num_simd_registers;
 static int num_opmask_registers;
 
+#ifndef DR_HOST_NOT_TARGET
+
+const static int num_feature_registers = sizeof(features_t) / sizeof(uint64);
+
+#    define MRS(REG, IDX)                                                            \
+        do {                                                                         \
+            if (IDX > (num_feature_registers - 1))                                   \
+                CLIENT_ASSERT(false, "Reading undefined AArch64 feature register!"); \
+            asm("mrs %0, " #REG : "=r"(isa_features[IDX]));                          \
```

Comment:
Better to avoid this implicit parameter: maybe define this inside read_feature_regs()?  Or pass this in as an explicit argument.

Reply:
Done.